doc: add Refine tutorial

necatiozmen opened the pull request:
## What does this PR do?

This PR adds a tutorial and quick start guide to the website for Refine and Appwrite. It was done with the knowledge of @gewenyu99 

## Test Plan
I was not able to render Refine icons after adding them the project. I may need a help for this.

## Related PRs and Issues


### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

TGlide commented:
Hi! The workflow for icons is: stick the SVG in `src/icons/svg`, then run `pnpm run icons:generate`.

gewenyu99 commented on src/icons/output/aw-icon.svg:
Take a look at Thomas' comments for the icon workflow. Sorry this isn't documented anywhere! First time we're doing an integration here with the new docs.

necatiozmen commented:
> Hi! The workflow for icons is: stick the SVG in `src/icons/svg`, then run `pnpm run icons:generate`.

Hi @TGlide,

I've placed the SVG in the `src/icons/svg` directory and executed the `pnpm run icons:generate` command. You can see the commit here: https://github.com/appwrite/website/pull/350/commits/6ea7d1fa0885795625da6a610d6a5f41b7e27af9.

However, I'm encountering an issue where the Refine SVG isn't appearing on the tutorials page. Interestingly, it displays correctly on both the quick start and home pages.

gewenyu99 commented on src/routes/docs/tutorials/+page.svelte:
```diff
@@ -19,7 +19,8 @@
         apple: 'icon-apple',
-        flutter: 'icon-flutter'
+        flutter: 'icon-flutter',
+        refine: 'icon-refine'
```
Try `aw-icon-refine`?

necatiozmen replied:
Thanks, it worked. The PR is ready now.

gewenyu99 commented:
<img width="361" alt="Screenshot 2023-11-28 at 9 28 35 AM" src="https://github.com/appwrite/website/assets/29069505/ec0b300c-1e04-410b-80ab-8ea70dd4b1e4">
I wonder if this logo looks like it's the right size. Let me check with design.

gewenyu99 commented on src/routes/docs/tutorials/refine/step-2/+page.markdoc:
```diff
@@ -0,0 +1,41 @@
+
+To make this example more visual, we'll use the Ant Desing UI package which natively supported by Refine.
+
+{% info title="Note" %}
```
```suggestion
{% info title="Existing projects" %}
```

gewenyu99 commented:
![refine](https://github.com/appwrite/website/assets/29069505/b56dd513-8e08-4dfc-b405-3cd98d596008)

necatiozmen commented:
> ![refine](https://private-user-images.githubusercontent.com/29069505/286333352-b56dd513-8e08-4dfc-b405-3cd98d596008.svg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNDEyMTIsIm5iZiI6MTcwMTI0MDkxMiwicGF0aCI6Ii8yOTA2OTUwNS8yODYzMzMzNTItYjU2ZGQ1MTMtOGUwOC00ZGZjLWI0MDUtM2NkOThkNTk2MDA4LnN2Zz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI5VDA2NTUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3MzI1ZmVhNzkwMWEzNDg5MGZmOGQ5YjlmNGFmZGE0MzMwZWJlOGEwYzNjYTRmNGFiYjQzNmZjNjIwNzgxOWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sBLi2iJZJ0fm0899fojp58WcO34JC5iriK0W3Uq3jqY)
> 
> We added the same padding size for the refine icon as we do all our other icons in the aw-icons library, so they'll match in size :)

Thank you, I updated it and renders properly now.

necatiozmen commented on src/routes/docs/tutorials/refine/step-4/+page.markdoc:
```diff
@@ -0,0 +1,165 @@
+import { appwriteClient } from './utility';
+import { authProvider } from './authProvider';
+
+const App: React.FC = () => {
```
I'll update all code blocks indentations.

necatiozmen commented on src/routes/docs/tutorials/refine/step-6/+page.markdoc:
```diff
@@ -0,0 +1,396 @@
+
+const App: React.FC = () => {
+    return (
+        <BrowserRouter>
```
I've updated the indentations, I think it more readable now. Please review again.

gewenyu99 commented:
Hey, I'm gonna merge this into a branch and carry it forward myself. Thank you so much for your help! This is looking great!

necatiozmen commented:
> Hey, I'm gonna merge this into a branch and carry it forward myself. Thank you so much for your help! This is looking great!

Thanks you so much

gewenyu99 commented:
<img width="313" alt="Screenshot 2023-12-13 at 3 08 26 PM" src="https://github.com/appwrite/website/assets/29069505/b388b8b1-3918-495d-b237-cbb4513187a5">
For the record.
